src/textual/widgets/_button.py

```diff
@@ -203,6 +205,13 @@ def __init__(
         self.active_effect_duration = 0.3
         """Amount of time in seconds the button 'press' animation lasts."""
 
+    def get_content_width(self, container: Size, viewport: Size) -> int:
+        try:
+            return max([cell_len(line) for line in self.label.plain.splitlines()]) + 2
```

Comment:
Good catch. Might it be worth adding a snapshot test for buttons with multi-line labels?